pkg/logs/input/tailer/scanner.go

```diff
@@ -39,28 +43,32 @@ func New(sources []*config.IntegrationConfigLogSource, pp pipeline.Provider, aud
 		default:
 		}
 	}
+	if tailingLimit == 0 {
+		tailingLimit = defaultTailingLimit
+	}
 	return &Scanner{
-		sources: tailSources,
-		pp:      pp,
-		tailers: make(map[string]*Tailer),
-		auditor: auditor,
+		pp:           pp,
+		fileProvider: NewFileProvider(tailSources, tailingLimit),
+		tailers:      make(map[string]*Tailer),
+		auditor:      auditor,
 	}
 }
 
 // setup sets all tailers
 func (s *Scanner) setup() {
-	for _, source := range s.sources {
-		if _, ok := s.tailers[source.Path]; ok {
-			log.Println("Can't tail file twice:", source.Path)
+	files := s.fileProvider.NewFiles()
```

Comment:
`NewFiles` is confusing, what about `FilesToTail`?

Reply:
We are in the tailer package